controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -1097,23 +656,32 @@ private void closeConnection(Reply reply, RawClient client) {
      * @return true if reply is in the expected reply set for the given requestType or throw exception.
      */
     @SneakyThrows(ConnectionFailedException.class)
-    private Boolean transformReply(Reply reply,
-                                   RawClient client,
-                                   String qualifiedStreamSegmentName,
-                                   Class<? extends Request> requestType,
-                                   WireCommandType type) {
+    private void handleReply(Reply reply,
+                             RawClient client,
+                             String qualifiedStreamSegmentName,
+                             Class<? extends Request> requestType,
+                             WireCommandType type) {
         closeConnection(reply, client);
-        Set<Class<? extends Reply>> expectedReplies = this.EXPECTED_REPLIES.get(requestType);
+        Set<Class<? extends Reply>> expectedReplies = EXPECTED_SUCCESS_REPLIES.get(requestType);
+        Set<Class<? extends Reply>> expectedFailingReplies = EXPECTED_FAILING_REPLIES.get(requestType);
         if (expectedReplies != null && expectedReplies.contains(reply.getClass())) {
             log.info(reply.getRequestId(), "{} {} {}.", requestType.getSimpleName(), qualifiedStreamSegmentName,
                     reply.getClass().getSimpleName());
-            return true;
-        } else if (reply instanceof WireCommands.NoSuchSegment) {
-            throw new WireCommandFailedException(type, WireCommandFailedException.Reason.SegmentDoesNotExist);
+        } else if (expectedFailingReplies != null && expectedFailingReplies.contains(reply.getClass())) {
+            if (reply instanceof WireCommands.NoSuchSegment) {
```

Comment:
This method is important for us to log properly, as it handles the failures got from the segment store. Could you please add here necessary log messages to make it easier for us to find problems related to `SegmentHelper`?

Reply:
Done